pkg/cmd/grafana-cli/services/api_client.go

```diff
@@ -118,32 +169,23 @@ func sendRequest(client http.Client, repoUrl string, subPaths ...string) ([]byte
 	req.Header.Set("grafana-arch", runtime.GOARCH)
 	req.Header.Set("User-Agent", "grafana "+grafanaVersion)
 
-	if err != nil {
-		return []byte{}, err
-	}
-
-	res, err := client.Do(req)
-	if err != nil {
-		return []byte{}, err
-	}
-	return handleResponse(res)
+	return req, err
 }
 
-func handleResponse(res *http.Response) ([]byte, error) {
+func handleResponse(res *http.Response) (io.ReadCloser, error) {
 	if res.StatusCode == 404 {
-		return []byte{}, ErrNotFoundError
+		return nil, ErrNotFoundError
 	}
 
 	if res.StatusCode/100 != 2 && res.StatusCode/100 != 4 {
-		return []byte{}, fmt.Errorf("Api returned invalid status: %s", res.Status)
+		return nil, fmt.Errorf("Api returned invalid status: %s", res.Status)
 	}
 
-	body, err := ioutil.ReadAll(res.Body)
-	defer res.Body.Close()
-
 	if res.StatusCode/100 == 4 {
-		if len(body) == 0 {
-			return []byte{}, &BadRequestError{Status: res.Status}
+		body, err := ioutil.ReadAll(res.Body)
+		defer res.Body.Close()
```

Comment:
Seems better to me to close `res.Body` in the function that obtains `res`, but looks as if this function already closes `res.Body` before your PR.

Reply:
Yes, I did not change that. Do you want me to change it?

Reply:
I guess it might be better to leave that for another PR.